tests/unittest_qmail-send.c

```diff
@@ -0,0 +1,239 @@
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_no_free_slots)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+
+  assert_job_avail(job_avail(), 0);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_a_free_slot)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(2);
+
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_a_free_slot_for_index_zero)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(0);
+
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_a_free_slot_for_last_index)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(numjobs - 1);
+
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_open_slots_full)
+{
+  job_init_this_many(2);
+  occupy_all_jobs();
+
+  int index = job_open(1,2);
+
+  ck_assert_int_eq(index, -1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_open_slot_available)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(3);
+  unsigned long job_open_id = 6546;
+  int job_open_channel = 5;
+
+  int index = job_open(job_open_id,job_open_channel);
+
+  ck_assert_int_eq(index, 3);
+  assert_job_at_index(3, unavailable);
+  ck_assert_uint_eq(jo[3].id, job_open_id);
+  ck_assert_int_eq(jo[3].channel, job_open_channel);
+  ck_assert_int_eq(jo[3].numtodo, 0);
+  ck_assert_int_eq(jo[3].flaghiteof, 0);
+}
+END_TEST
+
+START_TEST(test_qmail_job_open_multiple_slots_available)
+{
+  job_init_this_many(7);
+  occupy_all_jobs();
+  release_job(2);
+  release_job(3);
+  unsigned long job_open_id = 7890;
+  int job_open_channel = 12;
+
+  int index = job_open(job_open_id,job_open_channel);
+
+  ck_assert_int_eq(index, 2);
+  assert_job_at_index(2, unavailable);
+  ck_assert_uint_eq(jo[2].id, job_open_id);
+  ck_assert_int_eq(jo[2].channel, job_open_channel);
+  ck_assert_int_eq(jo[2].numtodo, 0);
+  ck_assert_int_eq(jo[2].flaghiteof, 0);
+}
+END_TEST
+
+/*
+ * job_close() could be called multiple times, dropping refs below 0.
+ * It doesn't look like qmail-send ever does this.
+ * But if we understand refs, we'd prefer boolean: available XOR unavailable.
+ */
+START_TEST(test_qmail_job_close_decrements_available_slots_for_each_call)
+{
+  job_init_this_many(7);
+  assert_job_at_index(3, available);
+
+  job_close(3);
+  assert_job_at_index(3, -1);
+  job_close(3);
+  assert_job_at_index(3, -2);
+}
+END_TEST
+
+START_TEST(test_qmail_job_close_decrements_specified_full_slot)
+{
+  job_init_this_many(7);
+  occupy_all_jobs();
+  release_job(5);
+  release_job(6);
+  unsigned long job_open_id = 79;
+  int job_open_channel = 42;
+
+  int index = job_open(job_open_id, job_open_channel);
+
+  ck_assert_int_eq(index, 5);
+  assert_job_at_index(index, unavailable);
+  job_close(index);
+  assert_job_at_index(index, available);
+  job_close(index);
+  ck_assert_int_lt(jo[index].refs, available);
+}
+END_TEST
+
+TCase
+*qmail_send_test_job_allocations(void)
```

Comment:
I'm fine if you split the line for return value and function name, but the ```*``` is part of the return value then.

Reply:
Can you help provide more guidance on the change you are expecting here? Apologies for the ignorance.

Reply:
Either write

```c
TCase *qmail_send_test_job_allocations(void)
```

or

```c
TCase *
qmail_send_test_job_allocations(void)
```